web/auth/auth.go

```diff
@@ -477,11 +461,11 @@ func extractState(r *http.Request, key string) (redirect string, teacher bool) {
 	return url[1:], teacher
 }
 
-func extractSessionCookie(w *echo.Response) string {
+func extractToken(w http.ResponseWriter) string {
 	// Helper function that extracts an outgoing session cookie.
 	outgoingCookies := w.Header().Values(OutgoingCookie)
 	for _, cookie := range outgoingCookies {
-		if c := strings.Split(cookie, "="); c[0] == SessionKey {
+		if c := strings.Split(cookie, "="); c[0] == "auth" {
```

Comment:
Go 1.18 has a nice [strings.Cut](https://pkg.go.dev/strings#Cut) function that would be perfect for this.